pkg/util/annotations/annotations_test.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package annotations
+
+import "testing"
+
+func TestValidate(t *testing.T) {
+	cases := []struct {
+		s             string
+		expectCorrect bool
+		expectedKey   string
+	}{
+		{
+			s:             "metadata.annotations['foo']",
+			expectCorrect: true,
+			expectedKey:   "foo",
+		},
+		{
+			s:             "metadata.annotations['\\[let\\'s\"escape\\[some\\]characters']",
+			expectCorrect: true,
```

Comment:
You shoudl document that the validation does not actually validate that they key is a valid key, just that it parses to a valid string.

Reply:
Or should you validate that it is syntactically correct (call `IsQualifiedName()` from k8s.io/apimachinery/pkg/util/validation ?